src/Model/Casts/CastResolver.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Mongolid\Model\Casts;
+
+use Mongolid\Model\Casts\DateTime\DateTimeCast;
+use Mongolid\Model\Casts\DateTime\ImmutableDateTimeCast;
+use Mongolid\Model\Casts\Exceptions\InvalidCastException;
+
+class CastResolver
+{
+    private const DATE_TIME = 'datetime';
+    private const IMMUTABLE_DATE_TIME = 'immutable_datetime';
+
+    public static array $validCasts = [
+        self::DATE_TIME,
+        self::IMMUTABLE_DATE_TIME,
+    ];
+
+    public static function resolve(?string $cast): ?object
```

Comment:
```suggestion
    public static function resolve(?string $cast): ?CastInterface
```


Particularmente, eu tiraria o null das propriedades, deixaria essa responsabilidade para a classe que chama esse resolver, prefiro porque evita a verificação de null em vários lugares e deixa claro que a responsabilidade da classe é pegar uma chave e encontrar um resolver. 

```suggestion
    public static function resolve(string $cast): CastInterface
```

Reply:
Boa! Na primeira implementação a gente ainda não tinha extraído essa CastInterface, por isso que o retorno tava bem genérico. Mas com as evoluções que foram feitas, não faz mais sentido retornar `?object` mesmo não

Reply:
Quanto ao retorno null, ele ainda é necessário porque a maioria dos campos não vão estar mapeados pra um cast. Aí quem usa precisa saber se deve castar ou não, a forrma de saber isso é verificar se um Cast existe. 

O lado bom é que só 4 lugares que usam esse método resolve(e 2 ddeles tão no legado). Por enquanto o uso tá controlado.

Reply:
Se não me engano a ideia de retirar nullable é mais uma questão de responsabilidade.

Essa deve retornar um "caster", dado um identificador de caster.

Essa questão do nullable deveria ser algo resolvido no código "cliente", se não tem configuração de cast, não deveria chamar a classe.

Reply:
Certamente é uma ideia interessante pra buscar deixar esse CastResolver bem específico.  Posso rascunhar ela numa branch paralela. 

Mas não vejo grandes diferenças na complexidade do código do cliente entre: 
- cliente verificar se tem cast pro atributo em questão antes de chamar o resolver 
- OU: cliente chamar o Resolver e lidar com o retorno nulo caso o atributo não tenha cast

Pelo menos não agora enquanto os usos ainda estão bem controlados.

Reply:
@gabrielgomes94 

A parte que me incomoda é mais legibilidade. Imagina que você tá debugando o código onde usa a classe. Você vê que a variável ta indo nula pra dentro da classe, aí você precisa ver o que que acontece quando vai nulo pra descobrir que ela só retorna nulo, parece um trabalho desperdiçado, efetivamente a classe não faz nada nesse caso. Se tivesse um caster padrão pra quando vir nulo, não veria problema.

Além disso é mais um caso pra ter que testar dentro dessa classe.

Reply:
Concordo com o pessoal. Se você não pode aceitar null, então nem deveria dar essa possibilidade no parâmetro, ainda mais que você já sabe qual é o tipo e não é algo dinâmico que poderia vir qualquer coisa em um mixed ou sem tipo

Reply:
Exemplos de como deixar essa tratativa do nulo do lado da classe cliente.

![image](https://github.com/leroy-merlin-br/mongolid/assets/974859/d5aebc19-5cff-4fb0-8a8e-fb6c929aaf50)
![image](https://github.com/leroy-merlin-br/mongolid/assets/974859/f3cd4e89-39eb-4c6d-a3ff-62159c1f74f1)

Reply:
Implementei: https://github.com/leroy-merlin-br/mongolid/pull/220/commits/0fa4607f1c7a1c45c9c09bd8411843350ac6dbe7